packages/graph-explorer-proxy-server/node-server.js

```diff
@@ -192,43 +235,57 @@ async function fetchData(res, next, url, options, isIamEnabled, region, serviceT
       },
       body,
     };
-    const isIamEnabled = !!req.headers["aws-neptune-region"];
-    const region = isIamEnabled ? req.headers["aws-neptune-region"] : "";
-    const serviceType = isIamEnabled ? (req.headers["service-type"] ?? DEFAULT_SERVICE_TYPE) : "";
 
     fetchData(res, next, rawUrl, requestOptions, isIamEnabled, region, serviceType);
   });
 
-  // cancel sparql query
-  app.get("/sparql/cancel", async (req, res, next) => {
-    const rawUrl = `${req.headers["graph-db-connection-url"]}/sparql/status`;
+  // POST endpoint for Gremlin queries.
+  app.post("/gremlin", async (req, res, next) => {
+    // Gather info from the headers
+    const queryId = req.headers["queryid"];
+    const graphDbConnectionUrl = req.headers["graph-db-connection-url"];
     const isIamEnabled = !!req.headers["aws-neptune-region"];
     const region = isIamEnabled ? req.headers["aws-neptune-region"] : "";
     const serviceType = isIamEnabled ? (req.headers["service-type"] ?? DEFAULT_SERVICE_TYPE) : "";
 
-    const requestOptions = {
-      method: "POST",
-      headers: {
-        "Content-Type": "application/x-www-form-urlencoded",
-      },
-      body: `cancelQuery&queryId=${req.headers["queryid"]}&silent=true`,
-    };
-
-    fetchData(res, next, rawUrl, requestOptions, isIamEnabled, region, serviceType);
-  });
-
-  // POST endpoint for Gremlin queries.
-  app.post("/gremlin", async (req, res, next) => {
     // Validate the input before making any external calls.
     if (!req.body.query) {
       return res
         .status(400)
         .send({ error: "[Proxy]Gremlin: query not provided" });
     }
-    const body = req.headers["queryid"]
-      ? { gremlin: req.body.query, queryId: req.headers["queryid"] }
-      : { gremlin: req.body.query };
-    const rawUrl = `${req.headers["graph-db-connection-url"]}/gremlin`;
+
+    /// Function to cancel long running queries if the client disappears before completion
+    async function cancelQuery() {
+      if (!queryId) {
+        return;
+      }
+      proxyLogger.debug(`Cancelling request ${queryId}...`);
+      return await retryFetch(
+        new URL(`${graphDbConnectionUrl}/gremlin/status?cancelQuery&queryId=${encodeURIComponent(queryId)}`),
```

Comment:
Same as above, This will only work for Neptune, we should, at least, add another ticket to update the code where needed to support other Gremlin endpoints.
The is no standardised way to cancel a connection to a Gremlin endpoint, so we will need a switch in future (neptune / generic-gremlin), or something similar, to only have certain features (like this one) when neptune is selected.

Reply:
I've added issue #271 to cover this.